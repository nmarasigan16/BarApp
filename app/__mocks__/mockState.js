export default mockState = {
    username: 'keshavsaharia',
    profile: {
        name: 'Keshav',
        followers: 10,
        following: 15,
        public_repos: 4,
        avatar_url: 'blahblah.com',
        website: 'moomoo.com',
        email: 'crapito@gmail.com',
        bio: 'I eat cheese',
        login: 'keshavsaharia'
    },
    followStatus: 1,
    repos: [
        {
            name: 'cat',
            description: 'meow',
            owner: {
                login: 'hello'
            },
            id: 40
        },
    ],
    followers: [
        {
            name: 'geordie',
            avatar_url: 'google.com',
            id: 3,
        },
    ],
    following: [
        {
            name: 'coomeron',
            avatar_url: 'jimmyjohns.com',
            id: 5,
        },
    ],
    starredRepos: [
        {
            name: '',
            description: '',
            owner: {
                login: ''
            },
            id: 1
        }
    ]
}