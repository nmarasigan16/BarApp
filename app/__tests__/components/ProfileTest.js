import 'react-native';
import React from 'react';
import mockState from '../../__mocks__/mockState';
import Bio from '../../src/components/profile/bio/BioComponent';
import Followers from '../../src/components/profile/followers/FollowersComponent';
import Number from '../../src/components/profile/number/NumberComponent';
import Repos from '../../src/components/profile/repos/RepoComponent';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('renders correctly', () => {
    it('bio with all items', () => {
        const tree = renderer.create(
            <Bio
                bio={mockState.profile.bio}
                website={mockState.profile.website}
                email={mockState.profile.email}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('bio without website', () => {
        const tree = renderer.create(
            <Bio
                bio={mockState.profile.bio}
                website={''}
                email={mockState.profile.email}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('bio without bio', () => {
        const tree = renderer.create(
            <Bio
                bio={''}
                website={mockState.profile.website}
                email={mockState.profile.email}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('followers', () => {
        const tree = renderer.create(
            <Followers
                follow={mockState.profile.followers}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('number', () => {
        const tree = renderer.create(
            <Number
                label={"Percentage"}
                number={1000}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('repos', () => {
        const tree = renderer.create(
            <Repos
                repos={mockState.profile.public_repos}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
