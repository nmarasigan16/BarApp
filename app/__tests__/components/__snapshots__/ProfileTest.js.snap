// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly bio with all items 1`] = `
<View>
  <View
    style={
      Object {
        "margin": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Arial",
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Arial",
            },
            Object {
              "fontSize": 38.259443278751455,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Bio
        

      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Arial",
            },
            Object {
              "fontSize": 30.607554623001164,
            },
          ]
        }
      >
        I eat cheese
      </Text>
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "margin": 15,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Arial",
          },
          Object {
            "flex": 0,
            "margin": 3,
            "textAlign": "center",
          },
        ]
      }
    >
      Website: 
      moomoo.com
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Arial",
          },
          Object {
            "flex": 0,
            "margin": 3,
            "textAlign": "center",
          },
        ]
      }
    >
      Email: 
      crapito@gmail.com
    </Text>
  </View>
</View>
`;

exports[`renders correctly bio without bio 1`] = `
<View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "margin": 15,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Arial",
          },
          Object {
            "flex": 0,
            "margin": 3,
            "textAlign": "center",
          },
        ]
      }
    >
      Website: 
      moomoo.com
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Arial",
          },
          Object {
            "flex": 0,
            "margin": 3,
            "textAlign": "center",
          },
        ]
      }
    >
      Email: 
      crapito@gmail.com
    </Text>
  </View>
</View>
`;

exports[`renders correctly bio without website 1`] = `
<View>
  <View
    style={
      Object {
        "margin": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Arial",
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Arial",
            },
            Object {
              "fontSize": 38.259443278751455,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Bio
        

      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Arial",
            },
            Object {
              "fontSize": 30.607554623001164,
            },
          ]
        }
      >
        I eat cheese
      </Text>
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "margin": 15,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "Arial",
          },
          Object {
            "flex": 0,
            "margin": 3,
            "textAlign": "center",
          },
        ]
      }
    >
      Email: 
      crapito@gmail.com
    </Text>
  </View>
</View>
`;

exports[`renders correctly followers 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 20,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "Arial",
              },
              Object {
                "fontSize": 30.607554623001164,
                "margin": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          Followers
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "Arial",
              },
              Object {
                "fontSize": 45.911331934501746,
                "fontWeight": "bold",
                "textAlign": "center",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "Arial",
              },
              Object {
                "fontSize": 30.607554623001164,
                "margin": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          Following
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "Arial",
              },
              Object {
                "fontSize": 45.911331934501746,
                "fontWeight": "bold",
                "textAlign": "center",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly number 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "Arial",
        },
        Object {
          "fontSize": 30.607554623001164,
          "margin": 10,
          "textAlign": "center",
        },
      ]
    }
  >
    Percentage
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "Arial",
        },
        Object {
          "fontSize": 45.911331934501746,
          "fontWeight": "bold",
          "textAlign": "center",
        },
      ]
    }
  >
    1000
  </Text>
</View>
`;

exports[`renders correctly repos 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "margin": 10,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Arial",
            },
            Object {
              "fontSize": 30.607554623001164,
              "margin": 10,
              "textAlign": "center",
            },
          ]
        }
      >
        Public Repositories
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Arial",
            },
            Object {
              "fontSize": 45.911331934501746,
              "fontWeight": "bold",
              "textAlign": "center",
            },
          ]
        }
      >
        4
      </Text>
    </View>
  </View>
</View>
`;
